bentoml/_internal/io_descriptors/image.py

```diff
@@ -101,47 +102,56 @@ def __init__(self, pilmode: str = DEFAULT_PIL_MODE, mime_type: str = "image/jpeg
 
         self._mime_type = mime_type.lower()
         self._pilmode = pilmode
-
-        ext = mimetypes.guess_extension(self._mime_type)
-        self._format = PIL.Image.EXTENSION[ext]
+        self._mime_type = mime_type
+        self._ext = mimetypes.guess_extension(mime_type)
+        if not self._ext:
+            raise ValueError(f"{mime_type} is not supported by the Image IO")
 
     def openapi_request_schema(self) -> t.Dict[str, t.Any]:
         """Returns OpenAPI schema for incoming requests"""
 
     def openapi_responses_schema(self) -> t.Dict[str, t.Any]:
         """Returns OpenAPI schema for outcoming responses"""
 
-    async def from_http_request(self, request: Request) -> "PIL.Image":
+    async def from_http_request(self, request: Request) -> "PIL.Image.Image":
         content_type, _ = parse_options_header(request.headers["content-type"])
-        if content_type == b"multipart/form-data":
-            form = await request.form()
-            contents = await next(iter(form.values())).read()
-            return PIL.Image.open(io.BytesIO(contents))
-
-        if (
-            content_type.decode("utf-8").startswith("image/")
-            or content_type == self._mime_type
-        ):
-            return PIL.Image.open(io.BytesIO(await request.body()))
-
-        raise BentoMLException(
-            f"{self.__class__.__name__} should have `Content-Type: multipart/form-data`, got {content_type} instead"
+        mime_type = content_type.decode()
```

Comment:
ugh shouldn't we also accept processing content-type as `multipart/form-data` as well?

Reply:
We definitely should, @bojiang what's the reason for removing that?

Reply:
this code actually doesn't work in my testing, because PIL needs the extension to read the image bytes.

Reply:
user may define `Multipart(img=Image())` to support the `multipart-form` mime?